daemon/graphdriver/fuse-overlayfs/fuseoverlayfs.go

```diff
@@ -96,7 +95,7 @@ func Init(home string, options []string, idMap idtools.IdentityMapping) (graphdr
 	if err := idtools.MkdirAllAndChown(home, 0710, dirID); err != nil {
 		return nil, err
 	}
-	if err := idtools.MkdirAllAndChown(path.Join(home, linkDir), 0700, currentID); err != nil {
+	if err := idtools.MkdirAllAndChown(filepath.Join(home, linkDir), 0700, currentID); err != nil {
```

Comment:
Any reason specifically to pick `filepath`? 

I think we should do the reverse for these? (use `path.Join()`, as these files won't be used on Windows, so `path.Join()` would be the correct match for `/` based filesystems. (And _possibly) even save the overhead of detecting which one to use / normalisation steps) ?

Reply:
I suppose we could go either way -- we /know/ we're on Linux, so we don't have to worry about picking the correct separator. On the other hand, the docs claim that filepath is supposed to be used for concrete filesystem paths, and path for abstract paths (like URIs)... In addition, I'd be shocked if there was any overhead for detection since if I was building this API there would be conditional compilation involved.

Happy to re-post this with path package, however.

Reply:
Hmm.. yeah, looks like I need to re-read the docs more often; I still recalled the "use path for slash-separated, and filepath for cross-platform"; looks like they added the "filesystem" note; https://github.com/golang/go/commit/7465bfb1ea1a7ddfec9b267587ee9e6200514f3f and the "such as URIs" in https://github.com/golang/go/commit/2f7fbf8851e4c45f7e0d207836a3c7a97eaa1823